iwyu.cc

```diff
@@ -2566,10 +2582,12 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     return visitor_state_->preprocessor_info;
   }
 
-  void AddUsingDeclaration(const NamedDecl* target_decl,  // what's being used
-                           const UsingDecl* using_decl) {
-    visitor_state_->using_declarations.insert(make_pair(target_decl,
-                                                        using_decl));
+  void AddShadowDeclarations(const UsingDecl* using_decl) {
+    for (UsingDecl::shadow_iterator it = using_decl->shadow_begin();
```

Comment:
You could use a range for-loop here and make this much nicer:

    for (const UsingShadowDecl* shadow : using_decl->shadows()) {
       ...
    }